### DISCUSSION TITLE
Speeding up initial action

### DISCUSSION THREAD

**Discussion by zilladotexe** [ronengr/hass_nuki_bt#72] [Ideas]
The first action I invoke on the lock often seems quite slow, with subsequent actions being much quicker. I'm assuming this is due to the device being asleep and/or bluetooth being unconnected. The official Nuki app can suffer with the same thing. Would it be possible to perhaps add an additional control that could be used to ready the device? Say for example a 'query state' action. Users of the integration could then call this at a suitable time (e.g when they unlock their phone before leaving the house) so the lock responds quickly when they need to actually lock it shortly after. Or maybe an alternative, when the lock is woken up e.g by pressing the physical button on the device, the nuki_bt integration connects to the device making it ready for a command. Basically, I want to make sure that when I invoke a lock command when leaving the house, the lock responds quickly.

Thanks for the work on the integration btw.

**Comment by ronengr** [comment]
I will add a 'query state' action in the next version.

Currently the integration connects to the lock (to query the state) every time the lock indicates the state has changed.
Connecting to it every time the lock wakes up will drastically reduce battery life, as the lock wakes up all the time to send a beacon. The way the integration knows when to connect and query the state is by looking at that beacon to see if a specific bit is on, indication the state was changed.

btw, the connection itself doesn't take much time. the reason it takes so long is only because the device is asleep.
You can try to improve the response time by changing the locks power save mode (called "Energy mode" in the Nuki app)
